integrations/symfony-bundle.rst

```diff
@@ -0,0 +1,176 @@
+| | httplug.plugin.history         | | These services are not public and may only be used when configure |
+|                                  | | HttpClients or other services.                                    |
++----------------------------------+---------------------------------------------------------------------+
+
+\* *These services are always an alias to another service. You can specify your own service or leave the default, which is the same name with `.default` appended. The default services in turn use the service discovery mechanism to provide the best available implementation. You can specify a class for each of the default services to use instead of discovery, as long as those classes can be instantiated without arguments.*
+
+
+If you need a more custom setup, define the services in your application configuration and specify your service in the `main_alias` section. For example, to add authentication headers, you could define a service that decorates the service `httplug.client.default` with a plugin that injects the authentication headers into the request and configure `httplug.main_alias.client` to the name of your service.
+
+.. code-block:: yaml
+
+    httplug:
+        clients:
+            acme_client: # This is the name of the client
+            factory: 'httplug.factory.guzzle6'
+
+        main_alias:
+            client: httplug.client.default
+            message_factory: httplug.message_factory.default
+            uri_factory: httplug.uri_factory.default
+            stream_factory: httplug.stream_factory.default
+        classes:
+            # uses discovery if not specified
+            client: ~
+            message_factory: ~
+            uri_factory: ~
+            stream_factory: ~
+
+
+Configuration without auto discovery
+````````````````````````````````````
+
+By default we use Puli to auto discover factories. If you do not want to use auto discovery you could use the following configuration (Guzzle):
+
+.. code-block:: yaml
+
+    httplug:
+        classes:
+            client: Http\Adapter\Guzzle6\Client
+            message_factory: Http\Message\MessageFactory\GuzzleMessageFactory
+            uri_factory: Http\Message\UriFactory\GuzzleUriFactory
+            stream_factory: Http\Message\StreamFactory\GuzzleStreamFactory
+
+
+Configure your client
+`````````````````````
+
+You can configure your clients with some good default options. The clients are later registered as services.
+
+.. code-block:: yaml
+
+    httplug:
+        clients:
+            my_guzzle5:
+                factory: 'httplug.factory.guzzle5'
+                config:
+                    # These options are given to Guzzle without validation.
+                    defaults:
+                        base_uri: 'http://google.se/'
+                        verify_ssl: false
+                        timeout: 4
+                        headers:
+                            Content-Type: 'application/json'
```

Comment:
i would prefer to only use guzzle options here that can not be handled by a php-http plugin. things like verify_ssl or timeout make a lot of sense. base_uri and headers are essentially different ways to achieve the same thing as configuring the plugins, and thus i think we should rather promote the plugins for that. will also make the client configuration more portable.